### PR TITLE
Auction view prototype

### PR DESCRIPTION
**Pull request by ashfurrow**
This is still pretty rough, but the hard part (the Relay support) is done.

simulator screen shot oct 28 2017 at 4 51 19 pm

Requires related metaphysics change: https://github.com/artsy/metaphysics/pull/795

Also refactors `InfiniteScrollGrid` to move away from icky `queryKey` towards a function-oriented approach.

Paired with @orta on getting Relay to work.

Related to: https://github.com/artsy/auctions/issues/682

Skip New Tests.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx`
- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/SaleArtworksGrid-tests.tsx`
- `src/lib/Components/Sale/__tests__/Header-tests.tsx`
- `src/lib/Containers/__tests__/Sale-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "sync-colors": "cd externals/elan && git fetch && git checkout origin/master && cp components/lib/variables/colors.json ../../data",
     "sync-externals": "npm run-script sync-schema && npm run-script sync-colors",
     "sync-schema": "cd externals/metaphysics && git fetch && git checkout origin/master && yarn install && yarn dump-schema -- ../../data && rm -rf node_modules",
+    "sync-schema:localhost": "cd ../metaphysics && yarn dump-schema -- ../emission/data",
```
Comment (mzikherman) [inline package.json]: You can just run this from Metaphysics (https://github.com/artsy/metaphysics/blob/80829d09aaab8ff7c5c6d09f18f915c472a7b34d/package.json#L22), but I like this version too!

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx`
- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/SaleArtworksGrid-tests.tsx`
- `src/lib/Components/Sale/__tests__/Header-tests.tsx`
- `src/lib/Containers/__tests__/Sale-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json (hunk 1 above)
Comment (ashfurrow) [reply package.json]: Yeah, this was Orta's idea to make it even easier :)

**Comment by ashfurrow** [comment]
This is good for review+merge (it's not used in production yet, so there's no harm in having it `master`). I added some basic test scaffolds so that as we develop out the container/components, it's obvious where the tests go.

File: data/schema.graphql [hunk 2]
```diff
@@ -668,7 +668,7 @@ type ArtworkConnection {
 
 union ArtworkContext = ArtworkContextAuction | ArtworkContextFair | ArtworkContextPartnerShow | ArtworkContextSale
 
-type ArtworkContextAuction {
+type ArtworkContextAuction implements Node {
```
Comment (orta) [inline data/schema.graphql]: these changes got merged, right?

Comment (ashfurrow) [reply data/schema.graphql]: yessir: https://github.com/artsy/metaphysics/pull/795

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**